src/Discord/Parts/User/Member.php

```diff
@@ -170,21 +170,32 @@ public function moveMember($channel)
      *
      * @param Role|int $role The role to add to the member.
      *
-     * @return bool Whether adding the role succeeded.
+     * @return bool returns whether role was added successfully
      */
     public function addRole($role)
     {
+
         if ($role instanceof Role) {
             $role = $role->id;
         }
 
         // We don't want a double up on roles
         if (false !== array_search($role, (array) $this->attributes['roles'])) {
+
             return false;
         }
 
         $this->attributes['roles'][] = $role;
 
+
+        $this->http->put(
+            "guilds/{$this->guild_id}/members/{$this->id}/roles/{$role}"
+        )->then(function ()  {
```

Comment:
Resolve the promise if the HTTP request succeds, reject otherwise:

```php
$this->http->put(...)->then(
    \React\Partial\bind([$deferred, 'resolve']),
    \React\Partial\bind([$deferred, 'reject'])
);
```